src/editors/containers/LibraryContentEditor/hooks.test.js

```diff
@@ -0,0 +1,279 @@
+import { useDispatch } from 'react-redux';
+import { renderHook } from '@testing-library/react-hooks';
+
+import * as requests from './data/requests';
+import { actions } from '../../data/redux';
+import * as module from './hooks';
+import { RequestKeys } from '../../data/constants/requests';
+import modes from './constants';
+
+jest.mock('react-redux', () => ({
+  useDispatch: jest.fn(),
+}));
+
+jest.mock('./data/requests', () => ({
+  fetchV2Libraries: jest.fn(),
+  fetchV1Libraries: jest.fn(),
+  fetchV2LibraryContent: jest.fn(),
+}));
+
+jest.mock('../../data/redux', () => ({
+  actions: {
+    library: {
+      loadLibraries: jest.fn(),
+      initializeFromBlockValue: jest.fn(),
+      unloadLibrary: jest.fn(),
+      setLibraryId: jest.fn(),
+      setLibraryVersion: jest.fn(),
+      initializeSettings: jest.fn(),
+      setLibraryBlocks: jest.fn(),
+    },
+    requests: {
+      failRequest: jest.fn(),
+    },
+  },
+}));
+
+const dispatch = jest.fn();
+const error = 'mockError';
+
+describe('useLibraryHook', () => {
+  const selectedLibraryId = 'soMeId';
+  const version = 'verNum';
+  const blockValue = {
+    data: {
+      metadata: {
+        source_library_id: selectedLibraryId,
+        source_library_version: version,
+        manual: 'this becomes selected if true',
+        max_count: 'this becomes count',
+        allow_resetting_children: 'this becomes showReset',
+        candidates: 'canDIdates TupLes',
+        field: 'sOmEFIeld',
+      },
+    },
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('useEffect when block finishs loading', () => {
+    const v2Libraries = [
+      {
+        id: 'v2libid',
+        title: 'v2libtitle',
+      },
+    ];
+    const v1Libraries = [
+      {
+        display_name: 'myv1library',
+        library_key: 'v1id',
+      },
+    ];
+    it('should fetch v2 and v1 libraries when block finishes loading', () => {
```

Comment:
A blank line between `it` statements is somehow just essential for me. At some point I just start seeing a wall of text otherwise.